scripts/zones/Mount_Zhayolm/npcs/Bapokk.lua

```diff
@@ -4,31 +4,40 @@
 -- Handles access to Alzadaal Ruins
 -- !pos -20 -6 276 61
 -----------------------------------
+local ID = zones[xi.zone.MOUNT_ZHAYOLM]
+-----------------------------------
 ---@type TNpcEntity
 local entity = {}
 
 entity.onTrade = function(player, npc, trade)
-    if npcUtil.tradeHas(trade, xi.item.IMPERIAL_SILVER_PIECE) then
-        player:startEvent(163)
+    if
+        trade:getItemCount() == 1 and
+        trade:hasItemQty(xi.item.IMPERIAL_SILVER_PIECE, 1)
+    then
+        player:tradeComplete()
+        player:setPos(-20, 3.7, 316, 198) -- using the pos method until the problem below is fixed
+        -- player:startEvent(163) -- << this CS goes black at the end, never fades in
     end
 end
 
 entity.onTrigger = function(player, npc)
-    if player:getZPos() > -280 then
+    if player:getZPos() > 280 then
         player:startEvent(164) -- Ruins -> Zhayolm
     else
-        player:startEvent(162) -- Zhayolm -> Ruins
+        if player:hasKeyItem(xi.ki.CAPTAIN_WILDCAT_BADGE) then -- Zhayolm -> Ruins
+            player:messageSpecial(ID.text.YOU_HAVE_A_BADGE, xi.ki.CAPTAIN_WILDCAT_BADGE)
+            player:setPos(-20, 3.7, 316, 198)
+            -- player:startEvent(163)
+        else
+            player:startEvent(162)
+        end
     end
 end
 
 entity.onEventUpdate = function(player, csid, option, npc)
 end
 
 entity.onEventFinish = function(player, csid, option, npc)
-    if csid == 163 then
-        player:confirmTrade()
-        player:setPos(-20, -6, 0, 192) -- using the pos method until the problem below is fixed
-    end
```

Comment:
Please do not delete this. Place a comment instead

Reply:
I'm suspecting that just like it happened with Bhaflau NPCs, same thing is happening here: We need to zone into Alzadaal directly in onEventFinish